### PR TITLE
support typed variation values (FF-800)

### PR DESCRIPTION
**Pull request by leoromanovsky**
## motivation

provide typed `Get*Assignment` functions

## decisions

`GetNumericAssignment` returns a `double` - is this appropriate for C#?

## for reviewers

There are some compiler warnings - worth addressing?

### CODE REVIEW

File: dot-net-sdk/EppoClient.cs [hunk 1]
```diff
@@ -84,12 +114,7 @@ public class EppoClient
             // Ignore Exception
         }
 
-        return assignedVariation?.value.StringValue();
-    }
-
-    public string? GetAssignment(string subjectKey, string experimentKey)
-    {
-        return this.GetAssignment(subjectKey, experimentKey, new SubjectAttributes());
+        return assignedVariation?.typedValue;
```
Comment (aarsilv) [inline dot-net-sdk/EppoClient.cs:117]: 🔥

File: dot-net-sdk/EppoClient.cs [hunk 2]
```diff
@@ -109,7 +134,7 @@ public class EppoClient
     public EppoValue GetSubjectVariationOverride(string subjectKey, ExperimentConfiguration experimentConfiguration)
     {
         var hexedSubjectKey = Shard.GetHex(subjectKey);
-        return experimentConfiguration.overrides.GetValueOrDefault(hexedSubjectKey, new EppoValue());
+        return experimentConfiguration.typedOverrides.GetValueOrDefault(hexedSubjectKey, new EppoValue());
```
Comment (aarsilv) [inline dot-net-sdk/EppoClient.cs:137]: 🔥

File: dot-net-sdk/dto/EppoValue.cs [hunk 3]
```diff
@@ -33,9 +33,14 @@ public EppoValue(EppoValueType type)
...
-    public long LongValue()
+    public bool BoolValue()
     {
-        return long.Parse(value, NumberStyles.Integer);
+        return bool.Parse(value);
+    }
+
+    public double DoubleValue()
+    {
+        return double.Parse(value, NumberStyles.Number);
```
Comment (aarsilv) [inline dot-net-sdk/dto/EppoValue.cs:43]: :+1:

File: eppo-sdk-test/EppoClientTest.cs [hunk 4]
```diff
@@ -44,29 +44,72 @@ public void TearDown()
...
+                break;
+            case "string":
+                var stringExpectations = assignmentTestCase.expectedAssignments.ConvertAll(x => x.StringValue());
+                Assert.That(GetStringAssignments(assignmentTestCase), Is.EqualTo(stringExpectations));
+
+                break;
+        }
+    }
+
+    private static List<bool?> GetBoolAssignments(AssignmentTestCase assignmentTestCase)
```
Comment (aarsilv) [inline eppo-sdk-test/EppoClientTest.cs:76]: 🙌

File: eppo-sdk-test/EppoClientTest.cs [hunk 5]
```diff
@@ -44,29 +44,72 @@ public void TearDown()
...
+        {
+            return assignmentTestCase.subjectsWithAttributes.ConvertAll(subject => client.GetBoolAssignment(subject.subjectKey, assignmentTestCase.experiment,
+                subject.subjectAttributes));
+        }
+
+        return assignmentTestCase.subjects.ConvertAll(subject =>
+            client.GetBoolAssignment(subject, assignmentTestCase.experiment));
+    }
+
+    private static List<double?> GetNumericAssignments(AssignmentTestCase assignmentTestCase)
```
Comment (aarsilv) [inline eppo-sdk-test/EppoClientTest.cs:89]: 🙌

**Review by aarsilv — APPROVED** [review]
This is great! Approving with a potential simplification for optional parameters you can take or leave as you see fit!

File: dot-net-sdk/eppo-sdk.csproj [hunk 6]
```diff
@@ -9,6 +9,7 @@
     <PackageProjectUrl>https://github.com/Eppo-exp/dot-net-server-sdk</PackageProjectUrl>
     <RepositoryUrl>https://github.com/Eppo-exp/dot-net-server-sdk</RepositoryUrl>
     <PackageId>Eppo.Sdk</PackageId>
+    <Version>1.0.0</Version>
```
Comment (leoromanovsky) [inline dot-net-sdk/eppo-sdk.csproj:12]: @agamkrbit Is this how versions are set in .net library projects? https://learn.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props

Comment (leoromanovsky) [reply dot-net-sdk/eppo-sdk.csproj]: Bumped to 2.0.0 as it seems the current version is `1.0.0` by default https://www.nuget.org/packages/Eppo.Sdk

**Pull request merged**